engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1738,38 +1831,72 @@ public Class<?> visit(MethodCallExpr n, VisitArgs printer) {
         return calculateMethodReturnTypeUsingGenerics(scope, method, expressionTypes, parameterizedTypes);
     }
 
-    private void checkPyNumbaVectorizedFunc(MethodCallExpr n, Expression[] expressions, Class<?>[] expressionTypes) {
-        // numba vectorized functions return arrays of primitive types. This will break the generated expression
-        // evaluation code that expects singular values. This check makes sure that numba vectorized functions must be
-        // used alone (or with cast only) as the entire expression.
+    private void prepareVectorization(MethodCallExpr n, Expression[] expressions, Class<?>[] expressionTypes) {
+        // find the Python CallableWrapper
+        final QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        QueryScopeParam<?>[] params = queryScope.getParams(queryScope.getParamNames());
+
+        Optional<QueryScopeParam<?>> paramOptional =
+                Arrays.stream(params)
+                        .filter(p -> p.getValue() instanceof CallableWrapper && p.getName().equals(n.getNameAsString()))
+                        .findFirst();
+
+        if (paramOptional.isPresent()) {
+            QueryScopeParam<?> param = paramOptional.get();
+            CallableWrapper callableWrapper = (CallableWrapper) param.getValue();
+            try {
+                checkVectorizability(n, expressions, expressionTypes, callableWrapper);
+                callableWrapper.setVectorizable(true);
+                if (!callableWrapper.isVectorized()) {
+                    log.info().append("Python function:").append(n.getNameAsString()).append(" is auto-vectorizable")
+                            .endl();
```

Comment:
DHFormualColumn has the following.  Does something like that belong here?
```java
if (callableWrapper.isVectorizable() && !callableWrapper.isVectorized()) {
                callableWrapper = vectorizeCallable(callableWrapper.getPyObject());
            }
```

Reply:
I think it is better where it is now as the negative case is logged closeby in the exception case. But a comment here might help the reader understand where is being done?

Reply:
I'm generally confused about what happens where.  There is similar looking functionality in `QueryLanguageParser` and `DHFormulaColumn`, but it is not the same functionality.  It isn't obvious to me why functionality belongs in one of the two places rather than both.  I don't know how much code I would need to read to convince myself.  How certain are you that it just goes in just one place?

Reply:
It is true that the 'vectorizability' check (in the parser)  and the actual monkey-patching (in the DhFormulaColumn) take place in two places. But because we are skipping the code-generation step which can't handle vectorization for now and directly hack the state of DhFormulaColumn to achieve numba and DH vectorization, having the monkey-patching done in DHFormulaColumn isn't so bad as it doesn't require us to expand the result structure of the parser.